### ISSUE TITLE
Ability to use "long lasting" Kubernetes certificates

### ISSUE THREAD

**Issue by sk4zuzu** [epiphany-platform/epiphany#1302]
By design Kubernetes assumes all newly created certificates have expiration time set to 1 year. There is really no automatic way to overcome that and modify expiration time (https://github.com/kubernetes/kubeadm/issues/1505#issuecomment-482798539).

What can be done though is:

1. A combination of automatic certificate renewal (https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/kubeadm-certs/#automatic-certificate-renewal) and kubelet certificate rotation (https://kubernetes.io/docs/tasks/tls/certificate-rotation/) could be used (after upgrade to 1.18+).

2. Custom certificate authority (https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/kubeadm-certs/#custom-certificates) managed by ansible, where we could try ignoring the 1 year requirement (at the cost of decreased security) and set it to arbitrary value.

**Comment by toszo** [comment]
Consider to backport it to 0.5.x branch

**Comment by atsikham** [comment]
As another task requires a hotfix, this one is on hold.

**Comment by atsikham** [comment]
Certificates settings in epiphany's config were refactored and extended, example:
```yaml
---
kind: configuration/kubernetes-master
title: "Kubernetes Master Config"
name: default
provider: azure
specification:
  advanced:
    certificates:
      location: /etc/kubernetes/pki
      expiration_days: 800
      update: false
```

Cases need to be tested 'at least':
* deployment, applying configuration with different OS on different cloud providers, as new ansible module was used
* single-master and HA installations, including promotion to HA, as update certificates process a bit different in that cases
* `update: true` and `update:false` with different periods, including default
* with `update:false` expiration date of `apiserver` certificate is not changed
* when certificates are updated, they have the same structure and fields as before update:
```shell
# useful commands
* kubeadm alpha certs check-expiration

# certificates location - /etc/kubernetes/pki/
* openssl x509 -in apiserver.crt -text -noout

# certificates embed to config files also participate in update process
* grep 'client-certificate-data' /etc/kubernetes/controller-manager.conf | awk '{print $2}' | base64 -d | openssl x509 -text
```

**Comment by atsikham** [comment]
Blocked by #1593. It's not possible to use ansible `openssl_*` modules across all supported OS distributions.
Draft [PR](https://github.com/epiphany-platform/epiphany/pull/1591) with changes that can be merged only if `python-cryptography` is installed on target hosts.

These changes can be modified to work without `openssl_*` modules, with `shell`, but it's better to upgrade to python3 and use these ones, as it will become complex and similar issues may occur in future.

**Comment by przemyslavic** [comment]
- single machine, single-master and HA installations have been tested
- tested parameter values: `renew: true`, `renew: false` with different periods, including default
- new installations have been tested as well as running `epicli apply` for the second time after changing parameters
